src/components/PostPageDetail/index.js

```diff
@@ -582,6 +586,11 @@ const PostPageDetailIdComponent = (props) => {
             isBackButton={true}
             source={SOURCE_PDP}
             height={getHeightHeader()}
+            isFollow={item?.is_following_target || parentData?.is_following_target}
+            onPressFollUnFoll={() => {
+              followUnfollow(item);
+              setItem({...item, is_following_target: !item?.is_following_target});
+            }}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code introduces a follow/unfollow feature. The logic seems correct, but it's important to ensure that the `followUnfollow` function handles all possible error scenarios. Also, directly mutating the state like this can lead to unexpected behavior. It would be better to use a callback function with the previous state as an argument in the `setItem` function.
```diff
- setItem({...item, is_following_target: !item?.is_following_target});
+ setItem(prevItem => ({...prevItem, is_following_target: !prevItem?.is_following_target}));
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->